docs/source/security/cors.mdx

```diff
@@ -0,0 +1,155 @@
+```bash
+# Same origin
+http://example.com:8080/ <==> http://example.com:8080/
+
+# Different origin (difference in domain, protocol, and port)
+http://example.com:8080/ =X= https://example1.com:8081/
+```
+
+However, as we all know, the internet is an exciting place full of resources that can make websites better (importing images, extra fonts, making API calls, etc.).  Developers needed a new protocol to relax SOP and safely _share_ resources across different origins.
+
+Cross-Origin Resource Sharing is the mechanism that allows a web page to  _share_ resources across different origins. CORS provides an extra layer of protection by enabling servers and clients to define HTTP headers that specify _which_ external clients' scripts can access their resources.
+
+Note that both SOP and CORS are related to [browser security](https://developer.mozilla.org/en-US/docs/Web/Security#security-related_glossary_terms); neither prevents other kinds of software from requesting resources from your server.
+
+## Choosing CORS options for your project
+
+When thinking about configuring CORS for your application, there are two main options you might want to change: the origins that can access your server's resources and whether your server accepts user credentials (i.e., cookies) with requests.
+
+### Specifying origins
+
+CORS uses [specific HTTP request headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#the_http_response_headers) as part of its protocol, including [`Access-Control-Allow-Origin`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin). The `Access-Control-Allow-Origin` header (ACAO) enables a server to dictate which origins can use scripts to access that server's resources.
+
+Depending on what you are building, the origins you specify in your CORS configuration might need to change when you are ready to deploy your application.
+
+#### ⚠️ Applications on private networks
+If your browser is running your API on a private network (i.e., not on the public internet), **the default CORS behavior of Apollo Server is insecure**. We strongly recommend [specifying which origins](#configuring-cors-options-for-apollo-server) can access your server's resources.
+
+Otherwise, while your computer is on your private network, a script on any website could potentially make your browser talk to your private API. [Some browsers, such as Chrome, have features under development](https://wicg.github.io/private-network-access/) to solve this problem. But in the meantime, all servers on private networks should *always* specify origins in their CORS configuration.
+
+#### APIs that require cookies
+If your API needs to accept [cross-origin cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies) with requests, you must specify origins in your CORS configuration. See [Passing credentials with CORS](#passing-credentials-with-cors) for examples.
```

Comment:
```suggestion
If your API needs to accept [cross-origin cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies) with requests, you must specify origins in your CORS configuration. Failing to specify origins isn't a security hole here; it just will prevent your API from successfully providing the cookies. See [Passing credentials with CORS](#passing-credentials-with-cors) for examples.
```

(maybe you can word this better)

Reply:
I'm not sure that you got the "this isn't an urgent security fix, it's just a thing you need to do to make your app work" aspect...